File: app/src/main/java/com/example/hw02/Calculator.java
package com.example.hw02;

import java.io.Serializable;

public class Calculator implements Serializable {
    public enum Action {
        PLUS,
        MINUS,
        MULTIPLY,
        DIVIDE,
        EQUALS
    }

    public static final String ERROR = "Error";

    private Float mvArgument1;
    private Action mvAction;

    protected Float calculate(Float ivArg1, Float ivArg2) {
        if (mvAction != null && ivArg1 != null && ivArg2 != null) {
            switch (mvAction) {
                case PLUS:
                    return ivArg1 + ivArg2;
                case MULTIPLY:
                    return ivArg1 * ivArg2;
                case MINUS:
                    return ivArg1 - ivArg2;
                case DIVIDE:
                    return ivArg2 == 0f ? null : (ivArg1 / ivArg2);
            }
        }
        return null;
    }

    public String onActionBtn(Action ivAct, Float ivArgument2) {
        boolean lvClearInput = false;

        if (mvAction != null) {
            mvArgument1 = calculate(mvArgument1, ivArgument2);
        } else if (ivAct == Action.EQUALS) {
            mvArgument1 = ivArgument2;
        } else {
            mvArgument1 = ivArgument2;
            lvClearInput = true;
        }

        if (ivAct != Action.EQUALS) {
            mvAction = ivAct;
        } else {
            mvAction = null;
        }

        if (lvClearInput) {
            return "";
        } else if (mvArgument1 != null) {
            return mvArgument1.toString();
        } else {
            return ERROR;
        }
    }

}

File: app/src/main/java/com/example/hw02/Calculator.kt
package com.example.hw02

import java.io.Serializable

class Calculator : Serializable {
    enum class Action {
        PLUS, MINUS, MULTIPLY, DIVIDE, EQUALS
    }
    private var mvArgument1: Float? = null
    private var mvAction: Action? = null

    private fun calculate(ivArg1: Float?, ivArg2: Float?): Float? {
        if (mvAction != null && ivArg1 != null && ivArg2 != null) {
            return when (mvAction) {
                Action.PLUS -> ivArg1 + ivArg2
                Action.MULTIPLY -> ivArg1 * ivArg2
                Action.MINUS -> ivArg1 - ivArg2
                Action.DIVIDE -> if (ivArg2 == 0f) null else ivArg1 / ivArg2
                else -> null
            }
        }
        return null
    }

    fun onActionBtn(ivAct: Action, ivArgument2: Float?): String {
        var lvClearInput = false

        when {
            mvAction != null -> mvArgument1 = calculate(mvArgument1, ivArgument2)
            ivAct == Action.EQUALS -> mvArgument1 = ivArgument2
            else -> {
                mvArgument1 = ivArgument2
                lvClearInput = true
            }
        }
        mvAction = if (ivAct != Action.EQUALS) {
            ivAct
        } else {
            null
        }
        return when {
            lvClearInput -> ""
            mvArgument1 != null -> mvArgument1.toString()
            else -> ERROR
        }
    }

    companion object {
        const val ERROR = "Error"
    }
}